pkg/header/check.go

```diff
@@ -63,37 +63,54 @@ func listFiles(config *ConfigHeader) ([]string, error) {
 			}
 			files, err := doublestar.Glob(pattern)
 			if err != nil {
-				return fileList, err
+				return nil, err
 			}
 			localFileList = append(localFileList, files...)
 		}
 
-		var seen = make(map[string]bool)
+		seen := make(map[string]bool)
 		for _, file := range localFileList {
 			files, err := walkFile(file, seen)
 			if err != nil {
-				return fileList, err
+				return nil, err
 			}
 			fileList = append(fileList, files...)
 		}
 	} else {
+		var candidates []string
+
+		t, _ := repo.Worktree()
+		s, _ := t.Status()
+		for file := range s {
+			candidates = append(candidates, file)
+		}
```

Comment:
We can handle status in fine-grained. However, I just simply add all possible candidates and filter files deleted out, and duplicate them below:

```go
		seen := make(map[string]bool)
		for _, file := range candidates {
			if !seen[file] {
				seen[file] = true
				if _, err := os.Stat(file); err == nil {
					fileList = append(fileList, file)
				}
			}
		}
```